lib/samples/create_and_edit_geometries/create_and_edit_geometries.dart

```diff
@@ -0,0 +1,607 @@
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                    onTap: onTap,
+                  ),
+                ),
+                // Build the bottom menu.
+                buildBottomMenu(),
+              ],
+            ),
+            Visibility(
+              visible: _showEditToolbar,
+              // Build the editing toolbar.
+              child: buildEditingToolbar(),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a map with an imagery basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImageryStandard);
+    // Set the map to the map view controller.
+    _mapViewController.arcGISMap = map;
+    // Add the graphics overlay to the map view.
+    _mapViewController.graphicsOverlays.add(_graphicsOverlay);
+    // Configure some initial graphics.
+    _graphicsOverlay.graphics.addAll(initialGraphics());
+    // Set an initial viewpoint over the graphics.
+    _mapViewController.setViewpoint(
+      Viewpoint.fromCenter(
+        ArcGISPoint(
+          x: -9.5920,
+          y: 53.08230,
+          spatialReference: SpatialReference(wkid: 4326),
+        ),
+        scale: 5000,
+      ),
+    );
+    // Do some initial configuration of the geometry editor.
+    // Initially set the created vertex tool as the current tool.
+    setState(() => _selectedTool = _vertexTool);
+    _geometryEditor.tool = _vertexTool;
+    // Listen to changes in canUndo and canRedo in order to enable/disable the UI.
+    _geometryEditor.onCanUndoChanged
+        .listen((value) => setState(() => _geometryEditorCanUndo = value));
+    _geometryEditor.onCanRedoChanged
+        .listen((value) => setState(() => _geometryEditorCanRedo = value));
+    // Listen to changes in isStarted in order to enable/disable the UI.
+    _geometryEditor.onIsStartedChanged
+        .listen((value) => setState(() => _geometryEditorIsStarted = value));
+    // Listen to changes in the selected element in order to enable/disable the UI.
+    _geometryEditor.onSelectedElementChanged.listen(
+      (value) =>
+          setState(() => _geometryEditorHasSelectedElement = value != null),
+    );
+    // Set the geometry editor to the map view controller.
+    _mapViewController.geometryEditor = _geometryEditor;
+  }
+
+  void onTap(Offset localPosition) async {
+    // Only select existing graphics to edit if the geometry editor is not started i.e. editing is not already in progress.
+    if (_geometryEditor.isStarted) {
+      return;
+    }
```

Comment:
I think it's more fluttery to put this kind of guard statement up in the `build` method
```dart
onTap: _geometryEditorIsStarted ? onTap : null,
```